File: app/src/main/java/com/example/app_loja_de_bolos/cake_details/data/CakeDetailsRepository.java
package com.example.app_loja_de_bolos.cake_details.data;

import com.example.app_loja_de_bolos.cake_details.model.CakeDetails
import kotlinx.coroutines.flow.Flow

interface CakeDetailsRepository {
    suspend fun getCakeDetailsById(cakeName: String): Flow<CakeDetails?>
}

File: app/src/main/java/com/example/app_loja_de_bolos/cake_details/data/CakeDetailsRepository.kt
package com.example.app_loja_de_bolos.cake_details.data

import com.example.app_loja_de_bolos.cake_details.model.CakeDetails

interface CakeDetailsRepository {
    suspend fun fetchCakeDetails(cakeId: String, cakeType: String, cakeCategory: String): CakeDetails?
    suspend fun addItemToCart(cake: CakeDetails): Boolean?
}